README.md

```diff
@@ -1,2 +1,283 @@
+
+```gradle
+dependencies {
+	implementation 'com.github.fingerprintjs:fingerprint-android:1.0.0'
+	
+	// Add this line for java-project
+	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"	
+}
+```
+
+## API Reference
+
+### Device ID and Fingerprint
+
+The library operates with two entities. 
+
+1. `deviceId` - it is an ID of the device, based on system-provided IDs. Currently, it is based on Google Service Framework ID, and ANDROID_ID if the first is unavailable. It's strict, which means the uniqueness of the ID for every device. Also, it keeps the same after application reinstalling.
+2. `fingerprint` - it is a digital fingerprint of a device. Basically, it is a hash, calculated from a lot of available platform signals. `fingerprint` **is not a strict ID**, so there is a probability that two different devices will have the same fingerprint. Also, there is a probability, that the same device will have a different fingerprint in different moments due to system update, settings changing, etc.
+
+#### Why do we need them both?
+It is a good question, most of use-cases can be covered with the `deviceId`. But the `GSF_ID` and `ANDROID_ID`can be spoofed much easier, than device data such as CPU info or sensors list. 
+
+Another reason for using `fingerprint` is a possible restriction of access to `deviceId` due to privacy policy changes trend.
+
+
+
+### Initialization and usage
+
+Initialize the library and get `fingerprint` and `deviceId`.
+
+In Kotlin
+
+```kotlin
+
+// Initialization
+val fingerprinter =
+            FingerprinterFactory.getInitializedInstance(applicationContext)
+
+// Simple usage
+val fingerprint = fingerprinter.fingerprint()
+val deviceId = fingerprinter.deviceId()
+
+```
+
+In Java
+
+```java
+
+// Initialization
+Fingerprinter fingerprinter =
+                FingerprinterFactory.getInitializedInstance(getApplicationContext());
+                
+// Simple usage
+String fingerprint = fingerprinter.fingerprint();
+String deviceId = fingerprinter.deviceId();
+
+```
+### Advanced usage
+The additional entities, provided by the library are `BaseFingerprinter` and `Configuration`.
+
+#### BaseFingerprinter
+
+`fingerprint` can be calculated using different platform signals. They are combined in several groups:
+
+1. Hardware fingerprint. (e.g. CPU info, sensors list etc.)
+2. Os Build fingerprint (the information about current ROM, its version etc.)
+3. Device state fingerprint (the information about basic settings of the device)
+4. Installed applications fingerprint
+
+
+Each of them has a class, which is inherited from `BaseFingerprinter` class.
+
+```kotlin
+
+abstract class BaseFingerprinter<T> (
+    val version: Int
+) {
+    abstract fun calculate(): String
+    abstract fun rawData(): T
+}
+
+```
+
+So the full public API of the library is following:
+
+```kotlin
+interface Fingerprinter {
```

Comment:
this looks too complicated for me with so many functions. 
I think it's also confusing "fingerprinters" with "signal providers".

I would prefer a single fingerprinter that used some default set of signal providers (which can be overriden with a mask or a config).